### PR TITLE
THREESCALE-10797: Enable SSO logout when SSO is enforced

### PR DESCRIPTION
**Pull request by jlledom**
**What this PR does / why we need it**:

Previously we decided that users shouldn't be able to logout when there's SSO is enforced and there's only one SSO provider published.

See the Jira issue: https://issues.redhat.com/browse/THREESCALE-2795
And the PR: https://github.com/3scale/porta/pull/2155

Note the second scenario in the PR: `when only one SSO integration is configured and enforced`

However, **THREESCALE-2983** complains about this behavior. This PR ensures the user can't skip the login screen even when there's only one published provider and one active session on that provider.

**Which issue(s) this PR fixes** 

[THREESCALE-10797](https://issues.redhat.com/browse/THREESCALE-10797)

**Verification steps** 

1. Configure the admin portal to enable an RH SSO integration
   **Account settings -> Users -> SSO Integrations**. Ask me for credentials.
2. Log out.
3. Instead of returning to the dashboard, you should see the login screen.

**Screencast**:

[Screencast from 2024-02-19 12-16-52.webm](https://github.com/3scale/porta/assets/17052241/e89f9b1c-befd-40eb-871c-f732ef725d45)

### CODE REVIEW

**Review by josemigallas — APPROVED** [review]

File: test/integration/provider/sessions_controller_test.rb [hunk 1]
```diff
@@ -41,23 +41,9 @@ class Provider::SessionsControllerTest < ActionDispatch::IntegrationTest
     assert_redirected_to "http://example.net/?provider_id=#{account.id}&user_id=#{account.admin_user.id}"
   end
 
-  test "redirect users to SSO when there's only one authentication provider and enforce_sso is on" do
+  test "does not redirect users to SSO" do
```
Comment (akostadinov) [inline test/integration/provider/sessions_controller_test.rb:44]:
```suggestion
  test "does not redirect users to SSO even with enforce_sso and single provider" do
```

**Comment by jlledom** [comment]
@akostadinov I commited your nitpick, now I can't wait for the question.

**Comment by akostadinov** [comment]
strange I don't see my comment. 

app/views/provider/sessions/logged_out.html.erb

why remove

**Comment by jlledom** [comment]
> app/views/provider/sessions/logged_out.html.erb
> 
> why remove

Just a cleanup. I don't think that's being used from anywhere since the last edit to that template is from 2010 and so far I've never seen a logout screen in porta, it always redirects to the login screen.

**Review by akostadinov — APPROVED** [review]

**Review by akostadinov — APPROVED** [review]

**Pull request merged**